bitnami/elasticsearch/templates/data-svc.yaml

```diff
@@ -8,6 +8,9 @@ spec:
   type: ClusterIP
   publishNotReadyAddresses: true
   ports:
+    - name: http
+      port: 9200
+      targetPort: http
```

Comment:
idem

Reply:
Actually this port is already used by readiness and liveness [probes](https://github.com/irizzant/charts-1/blob/0d29428887f122512627ca744fce8a726c9f2254/bitnami/elasticsearch/templates/data-statefulset.yaml#L126), so there is traffic on it from other Pods (kubelet in this case).
Moreover having this port open gives the possibility to use REST API with these nodes too whenever needed.

Reply:
The probes are run locally (localhost), therefore the traffic never gets routed by the associated service.

On the other hand, why would you like to access the REST API from the outside on other node but the coordinating-only one?

I'm okay with adding the port to the pod definition (data-statefulset.yaml) but I don't see any advantage on adding it to the service.

Reply:
>The probes are run locally (localhost), therefore the traffic never gets routed by the associated service.

The probe traffic is generated by Kubelet, not inside the Pod itself even if the URL is a localhost one.